python_modules/libraries/dagster-components/dagster_components/core/defs_module.py

```diff
@@ -41,13 +46,32 @@ class ComponentConfig(BaseModel, HasSourcePositionAndKeyPath):
     attributes: Optional[Mapping[str, Any]] = None
 
     @classmethod
-    def from_path(cls, path: Path) -> Optional["ComponentConfig"]:
+    def from_path(cls, path: Path) -> Optional[Self]:
         if not path.exists():
             return None
         else:
             return parse_yaml_file_to_pydantic(cls, path.read_text(), str(path))
 
 
+class DefsConfigModel(ResolvableModel):
+    post_processors: Optional[Sequence[AssetPostProcessorModel]] = None
+
+
+@dataclass
+class ResolvedDefsConfig(ResolvedFrom[DefsConfigModel]):
+    post_processors: Annotated[
+        Optional[Sequence[AssetPostProcessor]], Resolver.from_annotation()
+    ] = None
+
+    @classmethod
+    def from_path(cls, path: Path) -> Optional[Self]:
+        if not path.exists():
+            return None
+        else:
+            loaded_model = parse_yaml_file_to_pydantic(DefsConfigModel, path.read_text(), str(path))
+            return resolve_model(loaded_model, cls, ResolutionContext.default())
```

Comment:
@alangenfeld I think this sort of utility should ideally be baked into the `resolved` framework.

The component case is a special / silly one where we nest the config inside of another file, so we haven't ran into the need for this abstraction, but realistically under "normal usage" of the standalone Resolved framework we would expect people to want to load directly from a path into a resolved object.

Not high priority ofc

Reply:
something like  `MyResolvedClass.load_from_file(yaml_path)` via the base class?